Flaky Test Task_CRUD in http

I've seen this one fail a couple times, seems to reproduce easily on my machine too:
```
--- FAIL: TestTaskService (5.62s)
    --- FAIL: TestTaskService/TransactionalTaskService (0.00s)
        --- FAIL: TestTaskService/TransactionalTaskService/Task_CRUD (0.13s)
            servicetest.go:215: task not found
FAIL
FAIL	github.com/influxdata/influxdb/http	6.525s
```
https://circleci.com/gh/influxdata/influxdb/49155?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification

I was also able to reproduce easily:
```
go test -count=10 -run TestTaskService ./http/...
```

--- a/http/task_test.go
+++ b/http/task_test.go
@@ -16,7 +16,6 @@ import (
 )
 
 func TestTaskService(t *testing.T) {
-	t.Skip("TODO(#14108): flaky test (https://github.com/influxdata/influxdb/issues/14108)")
 	t.Parallel()
 	servicetest.TestTaskService(
 		t,

--- a/kv/task.go
+++ b/kv/task.go
@@ -792,7 +792,10 @@ func (s *Service) deleteTask(ctx context.Context, tx Tx, id influxdb.ID) error {
 	if err := taskBucket.Delete(key); err != nil {
 		return ErrUnexpectedTaskBucketErr(err)
 	}
-	return nil
+
+	return s.deleteUserResourceMapping(ctx, tx, influxdb.UserResourceMappingFilter{
+		ResourceID: task.ID,
+	})
 }
 
 // FindLogs returns logs for a run.

